src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php

```diff
@@ -211,6 +227,28 @@ public function iShouldNotSeeShipmentWithOrderNumber(OrderInterface $order): voi
         );
     }
 
+    /**
+     * @Then I should see :amount :product units in the list
+     */
+    public function iShouldSeeUnitsInTheList(int $amount, ProductInterface $product): void
+    {
+        $shipmentUnitsFromResponse = $this->responseChecker->getResponseContent($this->client->getLastResponse())['units'];
+
+        $productUnitsCounter = 0;
+        foreach ($shipmentUnitsFromResponse as $shipmentUnitFromResponse) {
+            /** @var ShipmentUnitInterface $shipmentUnit */
+            $shipmentUnit = $this->shipmentUnitRepository->findOneBy(['id' => explode('/', $shipmentUnitFromResponse)[4]]);
```

Comment:
IMO you should call an API endpoint for more information instead of using repository